extensions/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIOConfig.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.kafka;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import io.druid.segment.indexing.IOConfig;
+
+import java.util.Map;
+
+public class KafkaIOConfig implements IOConfig
+{
+  private final String sequenceName;
+  private final KafkaPartitions startPartitions;
+  private final KafkaPartitions endPartitions;
+  private final Map<String, String> consumerProperties;
+  private final boolean useTransaction;
+
+  @JsonCreator
+  public KafkaIOConfig(
+      @JsonProperty("sequenceName") String sequenceName,
```

Comment:
this is an internal impl detail and I do not think it needs to be documented. the supervisor is the user-facing thing.